### PR TITLE
started work on icons section

### PR DESCRIPTION
**Pull request by lukasoppermann**
Once completed this fixes https://github.com/github/primer/issues/1630.

### Related
- move package docs to repos: https://github.com/primer/octicons/pull/916
- https://github.com/primer/design/issues/389

## Redirects
The following redirects are needed once this PR gets merged
- https://primer.style/octicons/ → https://primer.style/foundations/icons/
- https://primer.style/octicons/guidelines/design → https://primer.style/foundations/icons/design-guidelines
- https://primer.style/octicons/:icon-name → https://primer.style/foundations/icons/:icon-name

## Todo
- [x] ~`build:icons` needs to be run on build of docs~ the json from `@primer/octicons-react` can be used
- [ ] ping @rezrah once this PR has been merged to [add the redirects](https://github.com/primer/design/issues/389)
- [ ] Archive docs from octicons

## Problems
- [x] `blob-stream` throws errors in the individual icon pages (will try to replace with native stream)
- [x] `jsPDF` [creates an error](https://github.com/primer/design/actions/runs/4114367813/jobs/7101630964) when gatsby is built

### CODE REVIEW

File: src/components/icons.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    // add current icon to height array
+    acc[cur.height].push(cur)
+
+    return acc
+  }, {}), [results])
+
+  return (
+    <Box display={"grid"} gridGap={5}>
+      <TextInput
+        icon={SearchIcon}
```
Comment (emilybrick) [inline src/components/icons.js:35]: For some reason the search icon placement is looking broken, is this a known issue?
Screen Shot 2023-02-07 at 12 08 23 PM

File: content/foundations/icons/index.mdx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+title: Octicons
+---
+import {HeroLayout} from '@primer/gatsby-theme-doctocat'
+import Icons from '../../../src/components/icons'
+
+export default HeroLayout
```
Comment (emilybrick) [inline content/foundations/icons/index.mdx:7]: Should we use the HeroLayout here? Or just have a standard type heading on this page (Octicons) that says the current version 
Screen Shot 2023-02-07 at 12 11 41 PM

Comment (lukasoppermann) [reply content/foundations/icons/index.mdx:7]: I copied it over the way it was, but I think we don't need the hero.

File: src/components/icons.js (hunk 1 above)
Comment (colebemis) [reply src/components/icons.js]: Fixed ✅

File: content/foundations/icons/index.mdx (hunk 2 above)
Comment (colebemis) [reply content/foundations/icons/index.mdx]: Removed the hero ✅

**Review by colebemis — APPROVED** [review]
Thanks for doing this, @lukasoppermann!

This looks great. I made a few adjustments:
- Updated the site to fetch the Octicon data from `unpkg.com` so we can get the latest data every time we build. This means that our documentation doesn't rely on the `@primer/octicons` version in `node_modules` being up-to-date.
- Removed the hero and made a couple of styling updates.
- Renamed "Icon overview" to "Overview" in the nav. "Icon overview" felt redundant nested under the "Icons" item.

**Review by colebemis — APPROVED** [review]

**Review by emilybrick — APPROVED** [review]
This looks great! Thank you! ✨

**Pull request merged**